### PR TITLE
Adds `EnvGateway` to initialize Firebase Admin based on Env

### PR DESCRIPTION
**Pull request by ggirotto**

### CODE REVIEW

File: firebase/src/core/data/gateways/firebase-app.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
...
 
-// TODO: Properly initialize this, taking into consideration the environment
-process.env["FIRESTORE_EMULATOR_HOST"] = "localhost:8080";
-export const app = admin.initializeApp({ projectId: "test-rpoj" });
+export function initializeFirebase(options: { isLocalDevelopment: boolean }): admin.app.App {
+  if (options.isLocalDevelopment) {
+    process.env["FIRESTORE_EMULATOR_HOST"] = "localhost:8080";
+  }
+
+  return admin.initializeApp();
```
Comment (matuella) [inline firebase/src/core/data/gateways/firebase-app.ts:8]: Does all emulators run properly if `projectId` is not used?

File: firebase/src/core/presentation/provider.ts [hunk 2]
```diff
@@ -24,15 +26,18 @@ export default class Provider {
   }
 
   // eslint-disable-next-line @typescript-eslint/no-empty-function
-  private constructor() {}
+  private constructor() {
```
Comment (matuella) [inline firebase/src/core/presentation/provider.ts:29]: remove lint above

File: firebase/src/core/data/gateways/firebase-app.ts (hunk 1 above)
Comment (ggirotto) [reply firebase/src/core/data/gateways/firebase-app.ts]: You're right. I forgot that scripts depend from `projectId`. Updated.

File: firebase/src/core/data/gateways/firebase-app.ts [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+  }
+
+  const serviceAccount = env.firebaseServiceAccount;
+  return admin.initializeApp({
+    projectId: serviceAccount.projectId,
+    credential: admin.credential.cert({
+      clientEmail: serviceAccount.clientEmail,
+      projectId: serviceAccount.projectId,
+      privateKey: serviceAccount.privateKey,
+    }),
```
Comment (ggirotto) [inline firebase/src/core/data/gateways/firebase-app.ts:16]: The `credentials` initialization it's not required when running the `functions` environment. But when running scripts locally, we need to properly authenticate with Firebase. To support both scenarios, I suggest to we always initialize with Firebase Service Account metadata

**Review by matuella — APPROVED** [review]

**Pull request merged**